ironfish/src/consensus/verifier.ts

```diff
@@ -689,27 +694,26 @@ export class Verifier {
   }
 
   static verifyEvmBurns(transaction: Transaction, result: EvmResult): VerificationResult {
-    // TODO(jwp): handle native asset as output balance (will require decrypting notes and comparing)
+    if (!transaction.evm) {
+      return {
+        valid: false,
+        reason: VerificationResultReason.EVM_BURN_BALANCE_MISMATCH,
+      }
+    }
+
     const assetBalanceDeltas = new AssetBalances()
 
     for (const event of result.events) {
-      if (event.assetId.equals(Asset.nativeId())) {
-        return {
-          valid: false,
-          reason: VerificationResultReason.EVM_UNSHIELD_EVENT_NATIVE_ASSET,
```

Comment:
can we get rid of this `VerificationResultReason`?

Reply:
The way I understand the verifying we have two steps:
1. Transaction balancing in rust which incorporates `evm.publicIron` and `evm.privateIron` with the native asset spends and outputs.
2. In the verifier check that `evm.publicIron` matches the number of native asset unshields and `evm.privateIron` matches the number of native asset shields

These checks should be sufficient. Joe said this check was just there because the checks had not yet been implemented yet so we shouldn't need it

Reply:
Yup both right. We can get rid of it.